Compilation fails on fedora 34

compilation fails on fedora 34 with
```
[1/322] Building CXX object Macaulay2/e/CMakeFiles/M2-engine.dir/eigen.cpp.o
FAILED: Macaulay2/e/CMakeFiles/M2-engine.dir/eigen.cpp.o 
/usr/bin/c++  -I../../Macaulay2/e -I../../Macaulay2/d -I../../Macaulay2/c -IMacaulay2/d -I/usr/include/libxml2 -Iusr-host/include -I/usr/include/frobby -isystem /usr/include/flexiblas -isystem /usr/include/eigen3 -O2 -DNDEBUG -march=native -DNDEBUG -DOM_NDEBUG -DSING_NDEBUG -Wuninitialized -g3 -I/home/pzinn/M2/M2/BUILD/fedora/usr-host/include -I/home/pzinn/M2/M2/include -I/home/pzinn/M2/M2/BUILD/fedora/include -Wall -Wextra -Wfatal-errors -Wcast-qual -Wno-unused-parameter -Wno-attributes -Wno-cast-qual -Wno-sign-compare -Wno-unused-local-typedefs -Wno-deprecated-copy -Wno-unknown-pragmas -Wno-ignored-qualifiers -Wno-implicit-fallthrough -fopenmp -std=gnu++17 -MD -MT Macaulay2/e/CMakeFiles/M2-engine.dir/eigen.cpp.o -MF Macaulay2/e/CMakeFiles/M2-engine.dir/eigen.cpp.o.d -o Macaulay2/e/CMakeFiles/M2-engine.dir/eigen.cpp.o -c ../../Macaulay2/e/eigen.cpp
In file included from /home/pzinn/M2/M2/include/M2/math-include.h:31,
                 from ../../Macaulay2/e/eigen.cpp:3:
../../Macaulay2/e/mpreal.h: In copy constructor 'eigen_mpfr::mpreal::mpreal(const eigen_mpfr::mpreal&)':
../../Macaulay2/e/mpreal.h:582:27: error: no matching function for call to 'eigen_mpfr::mpreal::mpfr_srcptr(mpfr_srcptr)'
  582 |     mpfr_init2(mpfr_ptr(),mpfr_get_prec(u.mpfr_srcptr()));
      |                           ^~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
[6/322] Building CXX object Macaulay2/e/CMakeFiles/M2-engine.dir/montableZZ.cpp.o
ninja: build stopped: subcommand failed.
```
I suspect this is related to:
https://github.com/advanpix/mpreal/issues/7
but am not competent to fix myself.

thanks for looking into it.
I tried your fix Anton (at least on my work comp, which runs Fedora 33 with mpfr-4.1.0-6.fc33.x86_64), and now I get
```
[1/533] Building C object Macaulay2/d/CMakeFiles/M2-interpreter.dir/gmp_aux.c.o
FAILED: Macaulay2/d/CMakeFiles/M2-interpreter.dir/gmp_aux.c.o 
/usr/bin/cc -DNDEBUG -DOM_NDEBUG -DSING_NDEBUG -I../../Macaulay2/d -I../../Macaulay2/e -I../../Macaulay2/c -IMacaulay2/d -I/usr/include/libxml2 -Iusr-host/include -I../../Macaulay2/system -isystem /usr/include/flexiblas -O2 -DNDEBUG -march=native -Wuninitialized -g3 -I/home/pzinn/M2/M2/BUILD/fedora/usr-host/include -I/home/pzinn/M2/M2/include -I/home/pzinn/M2/M2/BUILD/fedora/include -Wall -Wextra -Wfatal-errors -Wcast-qual -Wno-unused-parameter -Wno-attributes -Wno-cast-qual -Wno-sign-compare -Wno-uninitialized -Wno-unused -Wno-frame-address -fopenmp -std=gnu11 -MD -MT Macaulay2/d/CMakeFiles/M2-interpreter.dir/gmp_aux.c.o -MF Macaulay2/d/CMakeFiles/M2-interpreter.dir/gmp_aux.c.o.d -o Macaulay2/d/CMakeFiles/M2-interpreter.dir/gmp_aux.c.o -c ../../Macaulay2/d/gmp_aux.c
In file included from /home/pzinn/M2/M2/include/M2/math-include.h:31,
                 from ../../Macaulay2/d/gmp_aux.c:3:
../../Macaulay2/d/gmp_aux.c: In function ‘mpfr_hash’:
../../Macaulay2/d/gmp_aux.c:20:12: error: expected expression before ‘::’ token
   20 |   if (0 != mpfr_sgn(x))
      |            ^~~~~~~~
```
I'll try on my fedora 34 laptop tonight.

on fedora 34 laptop:
```
[1/245] Building C object Macaulay2/e/CMakeFiles/M2-engine.dir/complex.c.o
FAILED: Macaulay2/e/CMakeFiles/M2-engine.dir/complex.c.o 
/usr/bin/cc -DNDEBUG -DOM_NDEBUG -DSING_NDEBUG -I../../Macaulay2/e -I../../Macaulay2/d -I../../Macaulay2/c -IMacaulay2/d -I/usr/include/libxml2 -Iusr-host/include -I/usr/include/frobby -isystem /usr/include/flexiblas -isystem /usr/include/eigen3 -O3 -DNDEBUG -march=native -Wuninitialized -g3 -I/home/pzinn/M2/M2/BUILD/fedora/usr-host/include -I/home/pzinn/M2/M2/include -I/home/pzinn/M2/M2/BUILD/fedora/include -Wall -Wextra -Wfatal-errors -Wcast-qual -Wno-unused-parameter -Wno-attributes -Wno-cast-qual -Wno-sign-compare -Wno-unused-local-typedefs -fopenmp -std=gnu11 -MD -MT Macaulay2/e/CMakeFiles/M2-engine.dir/complex.c.o -MF Macaulay2/e/CMakeFiles/M2-engine.dir/complex.c.o.d -o Macaulay2/e/CMakeFiles/M2-engine.dir/complex.c.o -c ../../Macaulay2/e/complex.c
In file included from /home/pzinn/M2/M2/include/M2/math-include.h:31,
                 from Macaulay2/d/engine-exports.h:20,
                 from ../../Macaulay2/e/complex.h:13,
                 from ../../Macaulay2/e/complex.c:3:
../../Macaulay2/e/complex.c: In function ‘mpfc_init_set’:
../../Macaulay2/e/complex.c:11:3: error: expected expression before ‘::’ token
   11 |   mpfr_init_set(result->re, a->re, GMP_RNDN);
      |   ^~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
[4/245] Building C object Macaulay2/e/CMakeFiles/M2-engine.dir/table.c.o
ninja: build stopped: subcommand failed.
```

--- a/M2/include/M2/math-include.h
+++ b/M2/include/M2/math-include.h
@@ -28,6 +28,7 @@
   #include <gmp.h>
   #endif
 
+  #define MPFR_USE_NO_MACRO
   #include <mpfr.h>
   #include <mpfi.h>
 

